src/ng2-smart-table/components/cell/cell-types/cell-textarea.component.ts

```diff
@@ -0,0 +1,26 @@
+import { Component, Output, EventEmitter, Input } from '@angular/core';
+
+import { Cell } from '../../../lib/data-set/cell';
+import { DefaultCellType } from './default-cell-type';
+
+@Component({
+  selector: 'cell-type-textarea',
+  template: `
+    <textarea [ngClass]="inputClass"
+              class="form-control"
+              [(ngModel)]="cell.newValue"
+              [name]="cell.getColumn().id"
+              [disabled]="!cell.getColumn().isEditable"
+              (click)="onClick($event)"
```

Comment:
Thanks a lot for your suggestions. I had to do all the above + update npm and node.
Btw everything compiles with no errors now so I am updating the PR.